html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Email.pm

```diff
@@ -79,6 +80,9 @@ sub code : Path : Args(2) {
             get_logger->info("Extending duration to $unregdate");
             node_modify($node_mac, unregdate => $unregdate);
             pf::activation::set_status_verified($GUEST_ACTIVATION, $code);
+            my $sponsor_type = pf::Authentication::Source::EmailSource->getDefaultOfType;
```

Comment:
We can't assume that anymore, we should find out what is the actual source that was used during the sponsor registration

Reply:
s/sponsor/email/ also